drivers/dma/dma_renesas_rz.c

```diff
@@ -48,9 +98,18 @@ struct dma_renesas_rz_data {
 	/* Dma context should be the first in data structure */
 	struct dma_context ctx;
 	struct dma_channel_data *channels;
+
+	uint32_t curr_channel;
```

Comment:
How would this work if multiple channels are producing interrupts at any moment? This seems suspicious to me.